### PR TITLE
add `rs.utils.weighted_pearsonr` and tests

### PR DESCRIPTION
**Pull request by kmdalton**
I wrote a function to compute weighted Pearson correlation coefficients and added it to `utils/stats.py`.

### CODE REVIEW

File: reciprocalspaceship/utils/stats.py [hunk 1]
```diff
@@ -81,3 +81,41 @@ def compute_redundancy(
...
     return mult.get_hkls(), mult["Count"].to_numpy(np.int32)
+
+
+def weighted_pearsonr(x, y, w):
+    """
+    Calculate a [weighted Pearson correlation coefficient](https://en.wikipedia.org/wiki/Pearson_correlation_coefficient#Weighted_correlation_coefficient).
+
+    Note
+    ----
+    x, y, and w may be arbitrarily batched. the correlation coefficient will be computed pairwise along the last axis.
```
Comment (JBGreisman) [inline reciprocalspaceship/utils/stats.py:92]: I don't quite understand the use of "batched" in this context. Perhaps an example with the corresponding array shapes would help?

File: tests/utils/test_weighted_pearsonr.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import numpy as np
+import pytest
+from scipy.stats import pearsonr
+
+import reciprocalspaceship as rs
+
+
```
[the remark this answers is not in the document]
Comment (kmdalton) [reply tests/utils/test_weighted_pearsonr.py:7]: i don't know any. i'm happy to incorporate any examples you can find.

File: reciprocalspaceship/utils/stats.py (hunk 1 above)
Comment (kmdalton) [reply reciprocalspaceship/utils/stats.py]: is this better?

File: tests/utils/test_weighted_pearsonr.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (JBGreisman) [reply tests/utils/test_weighted_pearsonr.py:7]: yep -- unsure. I'm ok with this as is given there isn't another reference implementation for the general case.

[the remark this answers is not in the document]
Comment (kmdalton) [reply tests/utils/test_weighted_pearsonr.py:7]: i can add a regression test. is that a good idea?

[the remark this answers is not in the document]
Comment (JBGreisman) [reply tests/utils/test_weighted_pearsonr.py:7]: Perhaps -- do you mean by including a test against the current implementation? Also, is useful to test against `scipy` using uniform weights that aren't `np.ones`? Any uniform array should be equivalent, right?

[the remark this answers is not in the document]
Comment (kmdalton) [reply tests/utils/test_weighted_pearsonr.py:7]: yes test against  the current implementation.

[the remark this answers is not in the document]
Comment (kmdalton) [reply tests/utils/test_weighted_pearsonr.py:7]: okay i added a test against non-one weights and a test against a current result. i'll merge after ci.

**Pull request merged**